idx/cassandra/cassandra.go

```diff
@@ -36,15 +35,31 @@ const TableSchema = `CREATE TABLE IF NOT EXISTS %s.metric_idx (
     AND compression = {'sstable_compression': 'org.apache.cassandra.io.compress.LZ4Compressor'}`
 
 var (
-	// metric idx.cassadra.add.ok is how many metrics are successfully being indexed
-	idxCasOk = stats.NewCounter32("idx.cassandra.add.ok")
-	// metric idx.cassandra.add.fail is how many failures were encountered while trying to index metrics
-	idxCasFail = stats.NewCounter32("idx.cassandra.add.fail")
-	// metric idx.cassandra.add is the duration of addititons to the cassandra idx
-	idxCasAddDuration = stats.NewLatencyHistogram15s32("idx.cassandra.add")
-	// metric idx.cassandra.delete is the duration of deletions from the cassandra idx
-	idxCasDeleteDuration = stats.NewLatencyHistogram15s32("idx.cassandra.delete")
-	errmetrics           = cassandra.NewErrMetrics("idx.cassandra")
+	// metric idx.cassadra.query-insert.ok is how many insert queries for a metric completed successfully (triggered by an add or an update)
+	statQueryInsertOk = stats.NewCounter32("idx.cassandra.query-insert.ok")
+	// metric idx.cassandra.query-insert.fail is how many insert queries for a metric failed (triggered by an add or an update)
+	statQueryInsertFail = stats.NewCounter32("idx.cassandra.query-insert.fail")
+	// metric idx.cassadra.query-delete.ok is how many delete queries for a metric completed successfully (triggered by an update or a delete)
+	statQueryDeleteOk = stats.NewCounter32("idx.cassandra.query-delete.ok")
+	// metric idx.cassandra.query-delete.fail is how many delete queries for a metric failed (triggered by an update or a delete)
+	statQueryDeleteFail = stats.NewCounter32("idx.cassandra.query-delete.fail")
+
+	// metric idx.cassandra.query-insert.wait is time inserts spent in queue before being executed
+	statQueryInsertWaitDuration = stats.NewLatencyHistogram12h32("idx.cassandra.query-insert.wait")
+	// metric idx.cassandra.query-insert.exec is time spent executing inserts (possibly repeatedly until success)
+	statQueryInsertExecDuration = stats.NewLatencyHistogram15s32("idx.cassandra.query-insert.exec")
+	// metric idx.cassandra.query-delete.exec is time spent executing deletes (possibly repeatedly until success)
+	statQueryDeleteExecDuration = stats.NewLatencyHistogram15s32("idx.cassandra.query-delete.exec")
+
+	// metric idx.cassandra.add is the duration of an add of one metric to the cassandra idx, including the add to the in-memory index, excluding the insert query
```

Comment:
that seems a little confusing: it is the duration of an add of a metric to the cassandra idx, excluding the insert query? what is it then, just the add to the in-memory index?